connectors/class-connector-editor.php

```diff
@@ -187,33 +186,34 @@ public function action_links( $links, $record ) {
 	}
 
 	/**
-	 * Retrieves data submitted on the screen, and prepares it for the appropriate context type
+	 * Retrieves data submitted on the screen, prepares it for the appropriate context type and logs the changes
 	 *
-	 * @action load-theme-editor.php
-	 * @action load-plugin-editor.php
+	 * @action wp_ajax_edit-theme-plugin-file
 	 */
 	public function get_edition_data() {
-		if (
-			(
-				isset( $_SERVER['REQUEST_METHOD'] )
-				&&
-				'POST' !== sanitize_text_field( $_SERVER['REQUEST_METHOD'] )
-			)
-			||
-			'update' !== wp_stream_filter_input( INPUT_POST, 'action' )
-		) {
+		$action         = wp_stream_filter_input( INPUT_POST, 'action' );
+		$request_method = wp_stream_filter_input( INPUT_SERVER, 'REQUEST_METHOD' );
+
+		if ( ( isset( $request_method ) && 'POST' !== $request_method ) || ( 'edit-theme-plugin-file' !== $action ) ) {
 			return;
 		}
 
+		$location   = null;
 		$theme_slug = wp_stream_filter_input( INPUT_POST, 'theme' );
+
 		if ( $theme_slug ) {
+			$location          = 'theme-editor.php';
 			$this->edited_file = $this->get_theme_data( $theme_slug );
 		}
 
 		$plugin_slug = wp_stream_filter_input( INPUT_POST, 'plugin' );
+
 		if ( $plugin_slug ) {
+			$location          = 'plugin-editor.php';
 			$this->edited_file = $this->get_plugin_data( $plugin_slug );
 		}
+
+		$this->log_changes( $location );
```

Comment:
<!-- potential concern, ask for clarification or investigation -->
**❓question**
If there's no `$location` here, should we log anything? I don't think so?

Reply:
Fixed that.